be/src/exec/spill/input_stream.cpp

```diff
@@ -476,13 +476,16 @@ StatusOr<InputStreamPtr> BlockGroupSet::build_ordered_stream(std::vector<BlockGr
                                                              RuntimeState* state, const SerdePtr& serde,
                                                              Spiller* spiller, const SortExecExprs* sort_exprs,
                                                              const SortDescs* sort_descs) {
-    std::vector<InputStreamPtr> streams;
-    for (const auto& group : block_groups) {
-        BlockReaderOptions read_options;
-        if (spiller->options().enable_buffer_read) {
+    BlockReaderOptions read_options;
+    if (spiller->options().enable_buffer_read && block_groups.size() > 0) {
+        size_t max_buffer_bytes = spiller->options().max_read_buffer_bytes / block_groups.size();
+        if (max_buffer_bytes > config::spill_read_buffer_min_bytes) {
```

Comment:
why not add the below code to disable the feature explicitly?
```
else {
read_options.enable_buffer_read = false;
}
```

Reply:
We don't need to explicitly disable buffer read; BlockOption does so by default.

Reply:
Maybe BlockOption will change its default behavior someday.